### PR TITLE
support alias connection strings for eventhub resources

### PR DESCRIPTION
**Pull request by neil-yechenwei**
fixes terraform-providers/terraform-provider-azurerm#6593

### CODE REVIEW

File: azurerm/helpers/azure/eventhub.go [hunk 1]
```diff
@@ -100,7 +100,13 @@ func EventHubAuthorizationRuleSchemaFrom(s map[string]*schema.Schema) map[string
 			Default:  false,
 		},
 
-		"primary_key": {
+		"alias_primary_connection_string": {
```
Comment (katbyte) [inline azurerm/helpers/azure/eventhub.go:103]: could we rename this to 
```suggestion
		"primary_connection_string_alias": {
```

File: azurerm/helpers/azure/eventhub.go [hunk 2]
```diff
@@ -100,7 +100,13 @@ func EventHubAuthorizationRuleSchemaFrom(s map[string]*schema.Schema) map[string
...
+			Type:      schema.TypeString,
+			Computed:  true,
+			Sensitive: true,
+		},
+
+		"alias_secondary_connection_string": {
```
Comment (katbyte) [inline azurerm/helpers/azure/eventhub.go:109]: and
```suggestion
		"secondary_connection_string_alias": {
```

**Comment by sebader** [comment]
> thanks for the PR. I was wondering if it would make more sense to. put alias on the end so the properties sort and read better? thanks!

While I understand your argument, the way @neil-yechenwei implemented it is following the official SDK's path: https://github.com/Azure/azure-sdk-for-go/blob/master/services/eventhub/mgmt/2017-04-01/eventhub/models.go#L226

what is the general guidance for the TF provider? Does it have its own naming strategy or does it try to follow the underlying APIs?

**Comment by neil-yechenwei** [comment]
@sebader , the goal is to make users better understand these properties.

File: azurerm/helpers/azure/eventhub.go (hunk 2 above)
Comment (neil-yechenwei) [reply azurerm/helpers/azure/eventhub.go:109]: updated

File: azurerm/helpers/azure/eventhub.go (hunk 1 above)
Comment (neil-yechenwei) [reply azurerm/helpers/azure/eventhub.go:103]: updated

**Comment by neil-yechenwei** [comment]
@katbyte , thanks for your comments. I've updated code.

**Comment by katbyte** [comment]
@sebader - we try to give properties the most clear accurate name we can, much like the portal often deviates from the SDK/APi we do as well. They are often poorly (and sometimes incorrectly!) named, and in this case putting it at the end allowed the propertied to group in documentation & schema provider a better UX all around.

File: website/docs/r/eventhub_namespace.html.markdown [hunk 3]
```diff
@@ -88,14 +88,20 @@ The following attributes are exported:
 The following attributes are exported only if there is an authorization rule named
 `RootManageSharedAccessKey` which is created automatically by Azure.
 
+* `alias_default_primary_connection_string` - The alias of the primary connection string for the authorization
+    rule `RootManageSharedAccessKey`, which is generated when disaster recovery is enabled.
+
+* `alias_default_secondary_connection_string` - The alias of the secondary connection string for the
```
Comment (katbyte) [inline website/docs/r/eventhub_namespace.html.markdown:94]: I think you missed on here?

File: website/docs/r/eventhub_namespace.html.markdown [hunk 4]
```diff
@@ -88,14 +88,20 @@ The following attributes are exported:
 The following attributes are exported only if there is an authorization rule named
 `RootManageSharedAccessKey` which is created automatically by Azure.
 
+* `alias_default_primary_connection_string` - The alias of the primary connection string for the authorization
```
Comment (katbyte) [inline website/docs/r/eventhub_namespace.html.markdown:91]: I think you missed on here?

File: website/docs/d/eventhub_namespace.html.markdown [hunk 5]
```diff
@@ -47,14 +47,20 @@ output "eventhub_namespace_id" {
 The following attributes are exported only if there is an authorization rule named
 `RootManageSharedAccessKey` which is created automatically by Azure.
 
-* `default_primary_connection_string` - The primary connection string for the authorization
+* `alias_default_primary_connection_string` - The alias of the primary connection string for the authorization
```
Comment (katbyte) [inline website/docs/d/eventhub_namespace.html.markdown:50]: I think you missed on here?

File: website/docs/d/eventhub_namespace.html.markdown [hunk 6]
```diff
@@ -47,14 +47,20 @@ output "eventhub_namespace_id" {
...
     rule `RootManageSharedAccessKey`.
 
-* `default_secondary_connection_string` - The secondary connection string for the
+* `alias_default_secondary_connection_string` - The alias of the secondary connection string for the
```
Comment (katbyte) [inline website/docs/d/eventhub_namespace.html.markdown:53]: I think you missed on here?

File: azurerm/internal/services/eventhub/eventhub_namespace_resource.go [hunk 7]
```diff
@@ -158,13 +158,19 @@ func resourceArmEventHubNamespace() *schema.Resource {
 				},
 			},
 
-			"default_primary_connection_string": {
+			"alias_default_primary_connection_string": {
```
Comment (katbyte) [inline azurerm/internal/services/eventhub/eventhub_namespace_resource.go:161]: I think you missed on here?

File: azurerm/internal/services/eventhub/eventhub_namespace_resource.go [hunk 8]
```diff
@@ -158,13 +158,19 @@ func resourceArmEventHubNamespace() *schema.Resource {
...
 				Type:      schema.TypeString,
 				Computed:  true,
 				Sensitive: true,
 			},
 
-			"default_secondary_connection_string": {
+			"alias_default_secondary_connection_string": {
```
Comment (katbyte) [inline azurerm/internal/services/eventhub/eventhub_namespace_resource.go:167]: I think you missed on here?

File: azurerm/internal/services/eventhub/eventhub_namespace_authorization_rule_data_source.go [hunk 9]
```diff
@@ -34,12 +34,19 @@ func dataSourceEventHubNamespaceAuthorizationRule() *schema.Resource {
 
 			"resource_group_name": azure.SchemaResourceGroupNameForDataSource(),
 
-			"listen": {
-				Type:     schema.TypeBool,
-				Computed: true,
+			"primary_connection_string_alias": {
```
Comment (katbyte) [inline azurerm/internal/services/eventhub/eventhub_namespace_authorization_rule_data_source.go:37]: these are not ordered correctly?

File: azurerm/helpers/azure/eventhub.go [hunk 10]
```diff
@@ -100,7 +100,13 @@ func EventHubAuthorizationRuleSchemaFrom(s map[string]*schema.Schema) map[string
 			Default:  false,
 		},
 
-		"primary_key": {
+		"primary_connection_string_alias": {
```
Comment (katbyte) [inline azurerm/helpers/azure/eventhub.go:103]: these are not ordered correctly?

**Comment by jackofallops** [comment]
Running acctests now

**Comment by sebader** [comment]
@neil-yechenwei quick question on the usage of this new feature: When I do not know whether an EventHub is geo-replicated, is it safe to do it this way now? I.e. will the alias be null if not geo-replicated?
````
value = data.azurerm_eventhub_authorization_rule.audit_sender.primary_connection_string_alias != null ? data.azurerm_eventhub_authorization_rule.audit_sender.primary_connection_string_alias : data.azurerm_eventhub_authorization_rule.audit_sender.primary_connection_string
````

**Pull request merged**